pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -2033,17 +2033,34 @@ var (
 	add = func(logger klog.Logger, queue *PriorityQueue, pInfo *framework.QueuedPodInfo) {
 		queue.Add(logger, pInfo.Pod)
 	}
+	expectNoError = func(what string, err error) {
+		if err != nil {
+			panic(fmt.Sprintf("Unexpected error from %s: %v", what, err))
+		}
+	}
 	addUnschedulablePodBackToUnschedulablePods = func(logger klog.Logger, queue *PriorityQueue, pInfo *framework.QueuedPodInfo) {
 		// To simulate the pod is failed in scheduling in the real world, Pop() the pod from activeQ before AddUnschedulableIfNotPresent() below.
-		queue.activeQ.Add(queue.newQueuedPodInfo(pInfo.Pod))
-		if p, err := queue.Pop(); err != nil || p.Pod != pInfo.Pod {
+		expectNoError("Add", queue.activeQ.Add(queue.newQueuedPodInfo(pInfo.Pod)))
```

Comment:
https://github.com/kubernetes/community/blob/master/sig-scheduling/CONTRIBUTING.md#technical-and-style-guidelines

Use `t.Fatal`

Reply:
I would have preferred that myself. But `t` is not passed down into these operations. I could change that, but then the diff becomes larger.

Reply:
Or did you mean not having a `expectNoError` function ([DAMP](https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests/11837973#11837973))?

I find `expectNoError` more readable than an if check. But if that's just me, then I can change to `if ... t.Fatal`.

Reply:
Both done (using `t.Fatal` and removal of `expectNoError`).

Reply:
Thanks! More importantly I wanted you to use `t.Fatal`.
But I also don't like adding "assert" libraries.